doc/CMakeLists.txt

```diff
@@ -0,0 +1,39 @@
+# Provide a local project and cmake_minimum_required calls
+# to permit to generate the documentation without configuring
+# the complete project
+
+cmake_minimum_required(VERSION 3.0)
+
+# This variable should contain the argument of the project call in the main CMakeLists.txt
+set(ORIGINAL_PROJECT_NAME "LibTemplateCMake")
+project(${ORIGINAL_PROJECT_NAME}Docs)
+
+include(GNUInstallDirs)
+
+find_package(Doxygen QUIET)
+
+if(DOXYGEN_FOUND)
+  # This variables are used inside the Doxyfile.in template
+  set(DOXYGEN_BUILD_DIR "${CMAKE_CURRENT_BINARY_DIR}" CACHE STRING "Build directory for Doxygen html output (absolute).")
+  set(DOXYGEN_INSTALL_DIR "${CMAKE_INSTALL_DATAROOTDIR}/doc/${ORIGINAL_PROJECT_NAME}" CACHE STRING "Installation directory for Doxygen html output (relative to CMAKE_INSTALL_PREFIX).")
+  get_filename_component(ORIGINAL_PROJECT_DIR "${CMAKE_CURRENT_SOURCE_DIR}" PATH)
+  set(DOXYGEN_SOURCE_DIR "${ORIGINAL_PROJECT_DIR}/src")
+  set(DOXYGEN_DOC_DIR "${CMAKE_CURRENT_SOURCE_DIR}")
+
+
+  # Generate the doxygen configuration file used by the
+  # document generation. This step permits to use cmake variable
+  # such as @ORIGINAL_PROJECT_NAME@ in the Doxyfile.in template
+  configure_file(${CMAKE_CURRENT_SOURCE_DIR}/Doxyfile.in ${CMAKE_CURRENT_BINARY_DIR}/Doxyfile)
+
+  # The dox target will generate the doxygen documentation in ${DOXYGEN_BUILD_DIR}
+  add_custom_target(dox COMMAND ${DOXYGEN_EXECUTABLE} ${CMAKE_CURRENT_BINARY_DIR}/Doxyfile
+                        COMMENT "Generating doxygen documentation in ${DOXYGEN_BUILD_DIR}")
+
+  # Note: the install will install the documentation if present,
+  # but it will not trigger a regeneration of the documentation if it is outdated
+  install(DIRECTORY ${DOXYGEN_BUILD_DIR}/html
```

Comment:
I understand that, it would be nice to have a pre-installation command that deletes and recompile doxygen documentation. Something like that may be done by convertin *or using), if possible, scripts like [this (bash)](https://github.com/robotology/how-to-document-modules/blob/gh-pages/doxygen/doc-compile.sh) or [this (powershell)](https://github.com/robotology/how-to-document-modules/blob/gh-pages/doxygen/doc-compile.ps1).

Reply:
I am not fully understanding your comment.

Reply:
I was just commenting your comment, trying to understand whether we can "force" the installation target to re-run the doxygen target before perfomring installation. I was wondering if running a script (or a CMake-translated version of it) could be of help 😄

Reply:
The point against that is that adding enforcing a generation of the doxygen documentation if the software is installed will mean that the installation would be extremely slower for all users that have doxygen installed, so I did not really liked this solution.

Reply:
I undrstand your point, it also true that if one wants to install the doc, you take the consequence of building it. If we do not want to enforce it and keep a spearated target for the doc, this is toatally fine for me. Maybe we can add a status message hinting that the installation process install the latest built doc without rebuilding it.

Reply:
We agreed F2F not to recompile dox during install and to leave it just as a custom target.